plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/DeleteResourceHandler.java

```diff
@@ -84,7 +94,7 @@ protected boolean shouldCancel(IStatus lastCompletedJobResult, int numberOfFaile
 	    };
```

Comment:
I suggest to create and schedule one job for the bunch of resources, if there are more than 1 of them. The job, for instance, may iterate the stream and (1) check if monitor is cancelled, (2) report to monitor begin of task, (3) set 'deleting' to true, (4) explicitly call job.doRun(monitor) (let's make it public), (5) without a job listener, check status and set 'deleting' to false if ok.